### PR TITLE
Support Astropy Quantity as radius arg for Registry SkyCoord Spatial constraint

### PR DESCRIPTION
**Pull request by duytnguyendtn**
Hi PyVO maintainers!

This PR adds a small quality of life improvement to the Spatial registry constraint, to allow an astropy quantity to be provided for the radius argument. Was implementing this downstream in Jdaviz, but figured it would be broadly useful here as well?

I've added a simple test to verify the unit conversion back to degrees, as well as testing for invalid units.

I'll add a changelog entry once a PR number is assigned.

Thanks!
Duy

### CODE REVIEW

File: pyvo/registry/rtcons.py [hunk 1]
```diff
@@ -719,9 +719,16 @@ def tomoc(s):
 
         elif len(geom_spec) == 2:
             if isinstance(geom_spec[0], SkyCoord):
+                # If radius given is astropy quantity, then convert to degrees
```
Comment (bsipocz) [inline pyvo/registry/rtcons.py:722]: Please add this case to the docstring, and possibly an example to the narrative docs

File: pyvo/registry/rtcons.py [hunk 2]
```diff
@@ -719,9 +719,16 @@ def tomoc(s):
...
+                if isinstance(geom_spec[1], u.Quantity):
+                    if geom_spec[1].unit.physical_type != 'angle':
+                        raise ValueError("Radius quantity is not of type dimensionless angle.")
```
Comment (bsipocz) [inline pyvo/registry/rtcons.py:725]: Please update this error message, there are non-dimensionless angle units that work here, too.

**Review by bsipocz — COMMENTED** [review]
Please update the error message as e.g. degree works there just as well as radians.

File: pyvo/registry/rtcons.py (hunk 2 above)
Comment (duytnguyendtn) [reply pyvo/registry/rtcons.py]: Whoops, thanks for that! Fixed in https://github.com/astropy/pyvo/pull/594/commits/c0c135746021d35a2fb7ec16603f2972c00caa59

File: pyvo/registry/rtcons.py (hunk 1 above)
Comment (duytnguyendtn) [reply pyvo/registry/rtcons.py:726]: Tried my best to write useful examples!

Docstring example implemented in: https://github.com/astropy/pyvo/pull/594/commits/be70c436e21324b5c3cac208774b4d5a93f25956

Narrative example implemented in: https://github.com/astropy/pyvo/pull/594/commits/d91c80a4ee89ee13aacce9425b0e61dd34cea7cd

How do those look?

**Comment by bsipocz** [comment]
I'll look into the (unrelated) CI failure separately. In the meantime, would you mind squashing the commits a bit to remove the incremental convergence done in the last ~4?

(you can also run test locally either using the same tox commands the CI does, or by running `pytest pyvo -R` and `flake8 pyvo`, those would smoke out most of these.

**Comment by duytnguyendtn** [comment]
Ahh does PyVO not usually squash PR commits at merge time? I'm admittedly reckless when it comes to my commits due to the assumption they'll get squashed anyways when it gets accepted. I'll clean up the commit history now!

**Comment by bsipocz** [comment]
> Ahh does PyVO not usually squash PR commits at merge time

No, we assume nice logical chunks for commits 😅. It takes some time to get used to, but comes second nature soon enough.  (and given that we have large PRs, I find it useful to have them merged in in a manageable sequence into `main`, it will make it easier to dig back history/bugs as opposed to have one gigantic commit on merge and then a likely messy commit history separately in a PR).

File: pyvo/registry/rtcons.py (hunk 1 above)
Comment (bsipocz) [reply pyvo/registry/rtcons.py:726]: Re narrative docs: There is value to list the expected outputs, so I'll add those to your example, as well as will remove the IGNORE_OUTPUT, as we use that only for cases where results are expected to chang often or are unpredictable.

Comment (bsipocz) [reply pyvo/registry/rtcons.py:726]: (ahh, never mind, I see that we already have that in the previous example)

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Thank you @duytnguyendtn!

**Comment by bsipocz** [comment]
Having a look at this, we can interpret this PR as a bugfix, as quantities should have been supported already. So I'll remilestone, which will also help the jdaviz folks.

**Pull request merged**